components/clickup/clickup.app.js

```diff
@@ -0,0 +1,205 @@
+const axios = require("axios");
+
+module.exports = {
+  type: "app",
+  app: "clickup",
+  propDefinitions: {
+    workspace: {
+      type: "string",
+      label: "Workspace",
+      async options() {
+        const workspaces = (await this.getWorkspaces()).teams;
+        return workspaces.map((workspace) => {
```

Comment:
I generally prefer a simple map like this to not use a function block:

```javascript
return workspaces.map((workspace) => ({
 label: workspace.name,
 value: workspace.id,
}));
```

Reply:
Updated